internal/commands/result_test.go

```diff
@@ -132,6 +133,192 @@ func TestResultsExitCode_OnPartialScan_PrintOnlyFailedScannersInfoToConsole(t *t
+	}
+	assert.Assert(t, results.TotalCount == 1, "SCS Scorecard results should be excluded for VS Code agent")
+
+	defer os.Remove(fileName + ".json")
+}
+
+func TestRunScsResultsShow_Other_AgentsShouldNotShowScsResults(t *testing.T) {
+	executeCommand := func(agent string) *wrappers.ScanResultsCollection {
+		clearFlags()
+		mock.Flag = wrappers.FeatureFlagResponseModel{Name: wrappers.SCSEngineCLIEnabled, Status: true}
+
+		_, err := executeRedirectedOsStdoutTestCommand(createASTTestCommand(),
+			"results", "show", "--scan-id", "SCS", "--report-format", "json", "--agent", agent)
+		assert.NilError(t, err)
+
+		file, err := os.Open(fileName + ".json")
+		if err != nil {
+			t.Fatalf("failed to open file: %v", err)
+		}
+		defer file.Close()
+
+		fileContents, err := io.ReadAll(file)
+		if err != nil {
+			t.Fatalf("failed to read file: %v", err)
+		}
+
+		var results wrappers.ScanResultsCollection
+		err = json.Unmarshal(fileContents, &results)
+		assert.NilError(t, err)
+		return &results
+	}
+
+	results := executeCommand("Jetbrains")
+	for _, result := range results.Results {
+		assert.Assert(t, result.Type != params.SCSScorecardType && result.Type != params.SCSSecretDetectionType, "SCS results should be excluded for other agents")
+	}
+	assert.Assert(t, results.TotalCount == 0, "SCS Scorecard results should be excluded")
+
+	defer os.Remove(fileName + ".json")
+}
+
+func TestRunWithoutScsResults_Other_AgentsShouldNotShowScsResults(t *testing.T) {
+	executeCommand := func(agent string) *wrappers.ScanResultsCollection {
+		clearFlags()
+		mock.Flag = wrappers.FeatureFlagResponseModel{Name: wrappers.SCSEngineCLIEnabled, Status: true}
+		mock.Flag = wrappers.FeatureFlagResponseModel{Name: wrappers.ContainerEngineCLIEnabled, Status: true}
+
+		_, err := executeRedirectedOsStdoutTestCommand(createASTTestCommand(),
+			"results", "show", "--scan-id", "MOCK", "--report-format", "json", "--agent", agent)
+		assert.NilError(t, err)
+
+		file, err := os.Open(fileName + ".json")
+		if err != nil {
+			t.Fatalf("failed to open file: %v", err)
+		}
+		defer file.Close()
+
+		fileContents, err := io.ReadAll(file)
+		if err != nil {
+			t.Fatalf("failed to read file: %v", err)
+		}
+
+		var results wrappers.ScanResultsCollection
+		err = json.Unmarshal(fileContents, &results)
+		assert.NilError(t, err)
+		return &results
+	}
+
+	results := executeCommand("Jetbrains")
+	for _, result := range results.Results {
+		assert.Assert(t, result.Type != params.SCSScorecardType && result.Type != params.SCSSecretDetectionType, "SCS results should be excluded for other agents")
+	}
+	assert.Assert(t, results.TotalCount == 7, "SCS Scorecard results should be excluded")
+
+	defer os.Remove(fileName + ".json")
+}
+
+func TestRunNilResults_Other_AgentsShouldNotShowAnyResults(t *testing.T) {
+	executeCommand := func(agent string) *wrappers.ScanResultsCollection {
```

Comment:
this is the exact same function as in the test above.
better to extract it to private function and use it from both of the tests